firedrake/slate/preconditioners.py

```diff
@@ -89,10 +89,44 @@ def initialize(self, pc):
         # Set up the functions for the original, hybridized
         # and schur complement systems
         self.broken_solution = Function(V_d)
-        self.broken_rhs = Function(V_d)
+        self.broken_residual = Function(V_d)
         self.trace_solution = Function(TraceSpace)
         self.unbroken_solution = Function(V)
-        self.unbroken_rhs = Function(V)
+        self.unbroken_residual = Function(V)
+
+        # Set up the KSP for the hdiv residual projection
+        hd_ksp = PETSc.KSP().create(comm=pc.comm)
+        hd_ksp.setOptionsPrefix(prefix + "hdiv_residual_")
+
+        # HDiv mass operator
+        p = TrialFunction(V[self.vidx])
+        q = TestFunction(V[self.vidx])
+        mass = ufl.dot(p, q)*ufl.dx
+        # TODO: Bcs?
```

Comment:
I agree, something will probably need to be done. I'll leave this for another PR, as I currently raise a `NotImplementedError` in the presence of strong BCs. Once I hook up BCs, I think during assemble, we simply need to pass through `bcs=self.cxt.row_bcs`. Does that seem reasonable?